controllers/argocd/keycloak.go

```diff
@@ -1392,9 +1398,8 @@ func (r *ReconcileArgoCD) reconcileKeycloakForOpenShift(cr *argoprojv1a1.ArgoCD)
 		}
 	}
 
-	// If Keycloak deployment exists and a realm is already created for ArgoCD, Do not create a new one.
-	if existingDC.Status.AvailableReplicas == expectedReplicas &&
-		existingDC.Annotations["argocd.argoproj.io/realm-created"] == "false" {
+	// Proceed with the keycloak configuration only once the keycloak pod is up and running.
+	if existingDC.Status.AvailableReplicas == expectedReplicas {
```

Comment:
There is no major change here .

Previously, we only reconcile Keycloak configuration and update the OIDC configuration once(for the very first time when keycloak instance is created). But now, with the addition of `rootCA` property users should be able to add/remove/update the OIDC configuration at any given time.